### PR TITLE
Commands for Minecraft Server

### PR DESCRIPTION
**Pull request by JamesDearlove**
An upgraded and improved version of #40 

`/mcwhitelist` will let a UQCS member add one user to the whitelist. Admins override the one user limit
`/mcadmin run` allows people with correct permissions to manage the server directly. More can be added to this in the future so people don't have to remember specific commands.

Additionally, a new bot wide function to log directly to admin-alerts. Whitelisting will add when successful, and all runs of `mcadmin` are logged.

Required new environment variables for this to work:
`MC_RCON_ADDRESS` Address or IP address of the server with enabled RCON.
`MC_RCON_PORT` The port used for connecting to RCON.
`MC_RCON_PASSWORD` The password configured for RCON connection.

### CODE REVIEW

**Review by LimaoC — APPROVED** [review]
Looks good to me!

Once this is merged, we should probably clear the current whitelist as it's already got a decent number of players on it.

If a member wants to un-whitelist/change the name on the whitelist, I imagine they'd just contact committee for that? An un-whitelist command would probably be pretty abusable.

Also at some point we'll likely lock this command behind some sort of verified UQCS member role.

**Comment by JamesDearlove** [comment]
We can add some admin commands to remove people from the whitelist in the future, thought it would be more beneficial to get this up and running now.

File: uqcsbot/minecraft.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        return response
+
+    @app_commands.command()
+    @app_commands.describe(username="Minecraft username to whitelist.")
+    async def mcwhitelist(self, interaction: discord.Interaction, username: str):
+        """ Adds a username to the whitelist for the UQCS server. """
+        db_session = self.bot.create_db_session()
+        query = db_session.query(MCWhitelist).filter(MCWhitelist.discord_id == interaction.user.id)
+        is_user_admin = interaction.user.guild_permissions.manage_guild
```
Comment (LimaoC) [inline uqcsbot/minecraft.py:50]: There is a slight caveat to this, `interaction.user` can either be of type `User` or `Member`, and only `Member` has the `guild_permissions` field. I *think* this is to do with whether you send the bot a command in the server or not, as it won't be able to check your permissions without knowing which server you're talking about.

Comment (JamesDearlove) [reply uqcsbot/minecraft.py:50]: Ah good catch, I'll put a check in for this at some point

**Comment by andrewj-brown** [comment]
mc server hella dead, is it even worth adding this to the bot when no-one uses it?

**Pull request merged**